types/aws-lambda/trigger/api-gateway-proxy.d.ts

```diff
@@ -4,9 +4,32 @@ import {
 } from "../common/api-gateway";
 import { Callback, Handler } from "../handler";
 
+/**
+ * Works with Lambda Proxy Integration for Rest API or HTTP API integration Payload Format version 1.0
+ * @see - https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+ */
 export type APIGatewayProxyHandler = Handler<APIGatewayProxyEvent, APIGatewayProxyResult>;
+/**
+ * Works with Lambda Proxy Integration for Rest API or HTTP API integration Payload Format version 1.0
+ * @see - https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+ */
 export type APIGatewayProxyCallback = Callback<APIGatewayProxyResult>;
 
+/**
+ * Works with HTTP API integration Payload Format version 2.0
+ * @see - https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+ */
+export type APIGatewayProxyHandlerV2 = Handler<APIGatewayProxyEventV2, APIGatewayProxyResultV2>;
```

Comment:
I think these names should be `APIGatewayProxyV2*`, not `APIGatewayProxy*V2` to follow the pattern of "`FooHandler = Handler<FooEvent, FooResult>`, but I don't think there's any direct example of a v2 in this package beforehand so I'm happy to have this establish that semantically it's the whole set of types that are v2, not just its another set of types that happen to be v2 of existing ones... if that made any sense. Feedback welcomed!